### PR TITLE
System class

### PR DESCRIPTION
**Pull request by AlinaJansen**
Include system.py

### CODE REVIEW

File: Generators/GeneratorGeometries.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    INPUT: \\
+    --------------- 
+    inp: string, name of g96 file
+    ------------------------------
+    RETURN: \\
+    --------------- 
+    coords: list of coordinates
+    ------------------------------
+    '''
+    coords = []
```
Comment (FlorianAnders) [inline Generators/GeneratorGeometries.py:46]: 'coords' should be named 'list_coordinates'

File: Documentation/LaTeX/Documentation.tex
Comment (FlorianAnders) [inline Documentation/LaTeX/Documentation.tex:1]: Do we keep LaTeX files here or do we move to Overleaf?

File: stuff2sort/System.py [hunk 2]
```diff
@@ -0,0 +1,1150 @@
...
+#   NOTE:
+
+#   // CLASS & METHOD DEFINITIONS //
+class SystemInfo():
+
+    '''
+    This Class Reads And Stores Information About The System
+    '''
+
+    def __init__(self, input_dict) -> None:
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: 1) General and important note: we want to name variables the following way: datatype_specification_detailedspecification

As an example: 
instead of 'self.input_dict', it should be 'self.dict_input'; additionally, you can further specify which input, as we deal with a lot of different 'input types'.  So it would become 'self.dict_input_userparameters'. 

2) For function definitions, keep the practice of giving the return type. Here, you wrote 

def __init__(self, input_dict) -> None: 

which is the desired style, but in later function definitions you don't write the '-> <returned_datatype>' part. None is only to be used when no data is returned by the function; For example, if __init__ would return a list, you would write 

def __init__(self, input_dict) -> list:

3) More comments! Especially in physics / chemistry code it can become hard to follow multiple lines of code without a general description of specific steps taken

File: stuff2sort/System.py [hunk 3]
```diff
@@ -0,0 +1,1150 @@
...
+        RETURN: \\
+        --------------- \\
+        NONE \\
+        ------------------------------ \\
+        '''
+
+        self.input_dict = input_dict
+
+        #   Make a list of all topology files
+        self.topology_list = self.getincludelist(self.input_dict['topologyfile'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_topology_system OR self.list_topology

File: stuff2sort/System.py [hunk 4]
```diff
@@ -0,0 +1,1150 @@
...
+
+        # Read The Different Types Of Molecules In The System
+        self.list_of_molecules = self.read_molecules()
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_molecules is sufficient; leave out the 'of'.

File: stuff2sort/System.py [hunk 5]
```diff
@@ -0,0 +1,1150 @@
...
+
+        # Read The Charges For All Atoms
+        self.charge_vector = []
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_charges; Although a list in informatics could be called a vector, strictly speaking the datatype is a list. From a physics view, yes, this is a vector; however, we want to state the datatype. We could also cast the list into an np.array, making linear algebra easy to use, if necessary.

File: stuff2sort/System.py [hunk 6]
```diff
@@ -0,0 +1,1150 @@
...
+        for element in self.list_of_molecules:
+            self.charge_vector.extend(self.readcharges(element))
+
+        #   Read All Atom Lists
+        #   XX AJ currently I'm assuming we're always having separate files for atom indices, only remove this comment when we finally decided so or add the possibility for lists
+        self.qmatomslist = self.read_atoms_list(self.input_dict['qmatomslist'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Although the string is 'qmatomslist', the variable should be self.list_atoms_qm or similar.

File: stuff2sort/System.py [hunk 7]
```diff
@@ -0,0 +1,1150 @@
...
+        #   Make a list of all topology files
+        self.topology_list = self.getincludelist(self.input_dict['topologyfile'])
+
+        # Read The Different Types Of Molecules In The System
+        self.list_of_molecules = self.read_molecules()
+
+        # Read The Charges For All Atoms
+        self.charge_vector = []
+        for element in self.list_of_molecules:
+            self.charge_vector.extend(self.readcharges(element))
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Which datatype is element? 
Does element mean chemical element? Matrix element? Atom? Individual Molecule? 
(This is an example of ambigouous variable naming :D Could be confusing for anyone not familiar with the thought process / other functions)

File: stuff2sort/System.py [hunk 8]
```diff
@@ -0,0 +1,1150 @@
...
+
+        #   Read All Atom Lists
+        #   XX AJ currently I'm assuming we're always having separate files for atom indices, only remove this comment when we finally decided so or add the possibility for lists
+        self.qmatomslist = self.read_atoms_list(self.input_dict['qmatomslist'])
+        self.activeatomslist = self.read_atoms_list(self.input_dict['activeatomslist'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_atoms_active

File: stuff2sort/System.py [hunk 9]
```diff
@@ -0,0 +1,1150 @@
...
+        if input_dict['useinnerouter']:
+            self.inneratomslist = self.read_atoms_list(self.input_dict['inneratomslist'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_atoms_inner
self.list_atoms_outer

File: stuff2sort/System.py [hunk 10]
```diff
@@ -0,0 +1,1150 @@
...
+            self.outeratomslist = self.read_atoms_list(self.input_dict['outeratomslist'])
+        #   If we're not using inner/outer we're keeping the lists empty (currently necessary for some functions but that might change later)
+        else:
+            self.inneratomslist = []
+            self.outeratomslist = []
+
+        #   Read Connectivity
+        self.connectivity_list = self.read_connectity_from_topology(self.input_dict['topologyfile'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_connectivity_topology

File: stuff2sort/System.py [hunk 11]
```diff
@@ -0,0 +1,1150 @@
...
+            # Writing high-precision coordinate file
+            # logger(logfile, "Writing high-precision coordinate file...")
+            self.write_highprec(self.input_dict['coordinatefile']) # XX temp removed logfile until logfile decision of Florian AJ
+            self.input_dict['coordinatefile'] = self.input_dict['jobname'] + ".g96"
+            # logger(logfile, "Done.\n")
+        elif self.input_dict['coordinatefile'][-4:] == ".g96":
+            #logger(logfile, "Reading geometry (.g96)...\n")
+            self.initial_geometry = geometry.readg96(self.input_dict['coordinatefile'])
+
+        self.number_of_atoms = int(len(self.initial_geometry)/3)
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.int_number_atoms OR self.int_count_atoms OR similar

File: stuff2sort/System.py [hunk 12]
```diff
@@ -0,0 +1,1150 @@
...
+
+        #   Create xyzq (Coordinates And Charges) For The Whole System
+        #   XX AJ same here, I'll make one function of it
+        #   XX AJ technically, I would prefer this xyzq function not in this class, but it's used in the get_linkatoms_ang, so I'll keep it here
+        if input_dict['useinnerouter']:
+            self.xyzq = geometry.make_xyzq_io(self.initial_geometry, self.charge_vector, self.outeratomslist)
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: datatype of xyzq?

File: stuff2sort/System.py [hunk 13]
```diff
@@ -0,0 +1,1150 @@
...
+
+        #   Read Connectivity
+        self.connectivity_list = self.read_connectity_from_topology(self.input_dict['topologyfile'])
+
+        #   Read Initial Geometry
+        #   XX AJ I would prefer only one function here independent of the file type and only make that distinction within the function. I'll get back to that later when I'm writing GeneratorGeometries.py
+        if self.input_dict['coordinatefile'][-4:] == ".gro":
+            #logger(logfile, "Reading geometry (.gro)...\n")
+            # XX AJ if we rewrite gro files to g96 files, the following function can be deleted and we can read the g96 file with geometry.readg96 afterwards
+            self.initial_geometry = geometry.readgeo(self.input_dict['coordinatefile'])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.geometry_initial

File: stuff2sort/System.py [hunk 14]
```diff
@@ -0,0 +1,1150 @@
...
+        #   Create xyzq (Coordinates And Charges) For The Whole System
+        #   XX AJ same here, I'll make one function of it
+        #   XX AJ technically, I would prefer this xyzq function not in this class, but it's used in the get_linkatoms_ang, so I'll keep it here
+        if input_dict['useinnerouter']:
+            self.xyzq = geometry.make_xyzq_io(self.initial_geometry, self.charge_vector, self.outeratomslist)
+        else:
+            self.xyzq = geometry.make_xyzq(self.initial_geometry, self.charge_vector)
+
+        #   Read linkatoms and next order atoms
+        self.m1list = self.identify_m1()
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_atoms_m1
self.list_atoms_m2
 
I propose we use different names for m1 and m2; it is not directly obvious what m1 and m2 are.

File: stuff2sort/System.py [hunk 15]
```diff
@@ -0,0 +1,1150 @@
...
+        self.m2list = self.identify_m2()
+
+        self.linkatoms = get_linkatoms_ang(self.xyzq, self.qmatomslist, self.m1list, self.connectivity_list, [])
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_coordinates_linkatoms; 

I am not sure what ang in get_linkatoms_ang stands for, I presume angstrom? The unit does not matter here, but SHOULD be commented in the function before the return and / or here.

File: stuff2sort/System.py [hunk 16]
```diff
@@ -0,0 +1,1150 @@
...
+        self.linkcorrlist, self.q1list, self.q2list, self.q3list, self.m3list = self.get_linkcorrlist()
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: self.list_correlation_atoms_link (does corr stand for correlation? correction?)
self.list_coordinates_atoms_q1
self.list_coordinates_atoms_q2
self.list_coordinates_atoms_q3
self.list_coordinates_atoms_m3

File: stuff2sort/System.py [hunk 17]
```diff
@@ -0,0 +1,1150 @@
...
+
+
+
+
+    def read_atoms_list(self, atoms_input):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: def read_list_atoms_input

Return type missing

File: stuff2sort/System.py [hunk 18]
```diff
@@ -0,0 +1,1150 @@
...
+        --------------- \\
+        atoms_input: str -> Index File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        atom_list: list -> List Of Atom Indices \\
+        ------------------------------ \\
+        '''
+
+        atom_list = []
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: list_atoms

File: stuff2sort/System.py [hunk 19]
```diff
@@ -0,0 +1,1150 @@
...
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        ---------------
+        Reads Atom Indices
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        atoms_input: str -> Index File \\
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: file_input_atoms

File: stuff2sort/System.py [hunk 20]
```diff
@@ -0,0 +1,1150 @@
...
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        atom_list: list -> List Of Atom Indices \\
+        ------------------------------ \\
+        '''
+
+        atom_list = []
+        with open(atoms_input) as atom_file:
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Please specify the mode for open, i.e. 

with open(atoms_input, 'r') as ...

Where 'r' specifies the readonly mode.

File: stuff2sort/System.py [hunk 21]
```diff
@@ -0,0 +1,1150 @@
...
+            for line in atom_file:
+                if "[" in line or "]" in line:
```
Comment (FlorianAnders) [inline stuff2sort/System.py:148]: Comment about why we do this

File: stuff2sort/System.py [hunk 22]
```diff
@@ -0,0 +1,1150 @@
...
+        --------------- \\
+        NONE \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        molecule_list: list -> List Of Molecules And Their Amount In The System \\
+        ------------------------------ \\
+        '''
+
+        molecule_list = []
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: list_molecules

File: stuff2sort/System.py [hunk 23]
```diff
@@ -0,0 +1,1150 @@
...
+        with open(self.input_dict['topologyfile']) as ifile:
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: .... as file_input

File: stuff2sort/System.py [hunk 24]
```diff
@@ -0,0 +1,1150 @@
...
+            found = False
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: bool_match

File: stuff2sort/System.py [hunk 25]
```diff
@@ -0,0 +1,1150 @@
...
+            if not found:
+                #   XX AJ turn into Logging
+                print('No "molecules" entry in ' + str(self.top) + " found. Exiting.")
+                exit(1)
+            for line in ifile:
+                match = re.search(r"^;", line, flags=re.MULTILINE)
+                if match:
+                    continue
+                else:
+                    match = re.search(r"^(\S+)\s+(\d+)", line, flags=re.MULTILINE)
```
Comment (FlorianAnders) [inline stuff2sort/System.py:195]: Especially when working with Regex, try to formulate in a comment what you want to achieve with the specific regex (e.g. I want to extract lines where a non-whitespace character is found and followed by whitespaces and then digits), as regex is hard to read and one needs to remember or look up the symbolics. It helps to get an idea of the code more quickly. Additionally, if appropriate, try to formulate in a comment what you want to achieve with the regex search regarding the overall picture; Why do we want to find the lines with the specific regex? How do we want to treat them? Etc.

File: stuff2sort/System.py [hunk 26]
```diff
@@ -0,0 +1,1150 @@
...
+                        molecule_list.append([match.group(1), match.group(2)])
+                    else:
+                        #   XX AJ turn into Logging
+                        print("Found an incomprehensible line in molecules list. Exiting.")
+                        print("Last line was:")
+                        print(line)
+                        exit(1)
+        return molecule_list
+
+    def readcharges(self, molecule_entry):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: def read_charges_molecule

Return type missing

File: stuff2sort/System.py [hunk 27]
```diff
@@ -0,0 +1,1150 @@
...
+            for line in atom_file:
+                if "[" in line or "]" in line:
+                    continue
+                atom_index_list = re.findall("\d+", line)
+                if atom_index_list:
+                    atom_list.extend(map(int, atom_index_list))
+        atom_list = sorted(np.array(atom_list).astype(int))
+        return atom_list
+
+    def read_molecules(self):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Return type missing

File: stuff2sort/System.py [hunk 28]
```diff
@@ -0,0 +1,1150 @@
...
+                    flags=re.MULTILINE,
+                )
+                if match:
+                    charge_vector.append(float(match.group(1)))
+
+        charge_vector *= molecule_count
+
+        return charge_vector
+
+    def checkformol(self, molecule_name, topology_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: def check_occurence_topology_molecule

Return type missing

File: stuff2sort/System.py [hunk 29]
```diff
@@ -0,0 +1,1150 @@
...
+        topology_file: str -> Name Of The Topology File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        correct: bool \\
+        ------------------------------ \\
+        '''
+
+        with open(topology_file) as ifile:
+            correct = False
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: correct = False is funny :D 
And again an example for more readable (although admittedly longer) variable names;

bool_occurence_molecule

File: stuff2sort/System.py [hunk 30]
```diff
@@ -0,0 +1,1150 @@
...
+                            match = re.search(matchstring, line, flags=re.MULTILINE)
+                            if match:
+                                correct = True
+                            break
+                if correct:
+                    break
+
+        return correct
+
+    def read_connectity_from_topology(self, topology_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: def read_connectivity_topology would be sufficent here;

file_topology

Return type missing

File: stuff2sort/System.py [hunk 31]
```diff
@@ -0,0 +1,1150 @@
...
+        molecule_name: str -> Name Of The Molecule \\
+        topology_file: str -> Name Of The Topology File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        connectivity_list: list -> List Of The Connectivity Of All Molecules \\
+        ------------------------------ \\
+        '''
+
+        count = 0
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: int_count_

File: stuff2sort/System.py [hunk 32]
```diff
@@ -0,0 +1,1150 @@
...
+        connectivity_list = []
+        for element in self.list_of_molecules:
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: for molecule in self.list_mlecules (?)

File: stuff2sort/System.py [hunk 33]
```diff
@@ -0,0 +1,1150 @@
...
+    def readcharges(self, molecule_entry):
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        --------------- \\
+        Reads Charges For All Atoms In A Molecule Type \\
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        molvecentry: list -> List With Molecule Name And Amount Of This Molecule \\
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: molvecentry should be renamed

File: stuff2sort/System.py [hunk 34]
```diff
@@ -0,0 +1,1150 @@
...
+            current_topology = self.get_current_topology(element[0], topology_file)
+            for i in range(0, int(element[1])):
+                mollength = self.get_mollength_direct(element[0], current_topology)
+                connset = self.get_connlist(count, element[0], current_topology)
+                connectivity_list += connset
+                count += int(mollength)
+
+        return connectivity_list
+
+    def get_current_topology(self, molecule_name, topology_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_topology_current

Return type missing

File: stuff2sort/System.py [hunk 35]
```diff
@@ -0,0 +1,1150 @@
...
+        molecule_name: str -> Name Of The Molecule \\
+        topology_file: str -> Name Of The Topology File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        connectivity_list: list -> List Of The Connectivity Of All Molecules \\
+        ------------------------------ \\
+        '''
+
+        current_topology = topology_file
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: datatype_topology_current

File: stuff2sort/System.py [hunk 36]
```diff
@@ -0,0 +1,1150 @@
...
+        found = make_pcf.checkformol(molecule_name, topology_file)
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: bool_occurence_molecule

File: stuff2sort/System.py [hunk 37]
```diff
@@ -0,0 +1,1150 @@
...
+                    current_topology = element
+                    break
+        if not found:
+            # XX AJ replace with logger
+            print("No charges found for " + str(molecule_name) + ". Exiting.")
+            exit(1)
+
+        return current_topology
+
+    def get_mollength_direct(self, molecule_name, topology_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Return type missing

File: stuff2sort/System.py [hunk 38]
```diff
@@ -0,0 +1,1150 @@
...
+                                                        r"^\s*\n", line, flags=re.MULTILINE
+                                                    )
+                                                    if match:
+                                                        break
+                                            break
+                                break
+                    break
+        return mollength
+
+    def get_connlist(self, offset, molecule_name, topology_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_connectivity

Return type missing

File: stuff2sort/System.py [hunk 39]
```diff
@@ -0,0 +1,1150 @@
...
+                                                        r"^\s*\n", line, flags=re.MULTILINE
+                                                    )
+                                                    if match:
+                                                        break
+                                            break
+                                break
+                    break
+        return connectivity_list
+
+    def write_highprec(self, gro_file):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: write_file_gromacs_highprec

File: stuff2sort/System.py [hunk 40]
```diff
@@ -0,0 +1,1150 @@
...
+        gro_file: str -> Name Of The gro File \\
+        topology_file: str -> Name Of The Topology File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        filename: str -> Name Of The New g96 File \\
+        ------------------------------ \\
+        '''
+
+        # not tested yet, bc I didn't use a gro file AJ
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: I like that you give your Initials in the comments, but sometimes you put # XX AJ <Comment> and other times you put AJ at the end. Just a remark that even in comments it would be nice if you keep to the same style.

File: stuff2sort/System.py [hunk 41]
```diff
@@ -0,0 +1,1150 @@
...
+                                # logger(logfile,str("Unexpected line instead of box vectors. Exiting. Last line:\n"))
+                                # logger(logfile,line)
+                                exit(1)
+                        else:
+                                ofile.write(str(" {:>15.9f} {:>15.9f} {:>15.9f}\n".format(float(match.group(1)),float(match.group(2)),float(match.group(3)))))
+                        ofile.write("END")
+
+        return filename
+
+    def identify_m1(self):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_atoms_m1

Return type missing

File: stuff2sort/System.py [hunk 42]
```diff
@@ -0,0 +1,1150 @@
...
+        for element in self.qmatomslist:
+            bondlist = self.get_bondpartners(element)
+            for entry in bondlist:
+                if (int(entry) not in np.array(self.qmatomslist).astype(int)) and (
+                    int(entry) not in np.array(m1list).astype(int)
+                ):
+                    m1list.append(entry)
+        return m1list
+
+    def get_bondpartners(self, target):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_atoms_m1qm

Return type missing

File: stuff2sort/System.py [hunk 43]
```diff
@@ -0,0 +1,1150 @@
...
+                    for i in range(1, len(entry)):
+                        if int(entry[i]) not in np.array(partnerlist).astype(int):
+                            partnerlist.append(int(entry[i]))
+                else:
+                    if int(entry[0]) not in np.array(partnerlist).astype(int):
+                        partnerlist.append(int(entry[0]))
+
+        return partnerlist
+
+    def identify_m2(self):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_atoms_m2

File: stuff2sort/System.py [hunk 44]
```diff
@@ -0,0 +1,1150 @@
...
+        for element in self.m1list:
+            m2line = []
+            bondlist = self.get_bondpartners(element)
+            for entry in bondlist:
+                if int(entry) not in np.array(self.qmatomslist).astype(int):
+                    m2line.append(entry)
+            m2list.append(m2line)
+        return m2list
+
+    def get_linkcorrlist(self):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_atoms_link

Return type missing

File: stuff2sort/System.py [hunk 45]
```diff
@@ -0,0 +1,1150 @@
...
+            final_linkcorrlist.append(element)
+        return (
+            sorted(final_linkcorrlist, key=lambda l: l[0]),
+            q1list,
+            q2list,
+            q3list,
+            m3list,
+        )
+    
+    def getincludelist(self, topology_input):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: get_list_topologies

Return type missing

File: stuff2sort/System.py [hunk 46]
```diff
@@ -0,0 +1,1150 @@
...
+                        check = os.path.isfile(foundname)
+                        if not check:
+                            print("File " + foundname + " was not found. Maybe update the gmxpath variable in the script? Exiting.")
+                            exit(1)
+                    toplist.append(foundname)
+
+                    toplist.extend(self.getincludelist(foundname))
+        return toplist
+    
+    def get_atoms(self, qmmm_topology):
```
Comment (FlorianAnders) [inline stuff2sort/System.py]: Do we need this function? mass map could be stored in a JSON file for easy read up when necessary, or, if desired, easy change / addition of atom types

**Review by FlorianAnders — CHANGES_REQUESTED** [review]
Please check System.py for comments before our meeting

File: stuff2sort/System.py (hunk 40 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: AJ is for permanent comments, # XX AJ is for temporary comments, where one of us needs to do or decide something. That way we can search for XX to check if there's something left to discuss in the file.

File: stuff2sort/System.py (hunk 25 above)
Comment (AlinaJansen) [reply stuff2sort/System.py:195]: I understand the benefit, I just want to mention that it will take a long time to do that for the whole code. We can discuss that in the meeting today.

File: Documentation/LaTeX/Documentation.tex
Comment (AlinaJansen) [reply Documentation/LaTeX/Documentation.tex:1]: Discuss in meeting today

File: stuff2sort/System.py (hunk 2 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Understood (:

File: stuff2sort/System.py (hunk 3 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 4 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 5 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 6 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 7 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: changed to molecule (:

File: stuff2sort/System.py (hunk 8 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 9 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 10 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 11 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 12 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: xyzq is a numpy array. Is self.array_xyzq sufficient here? Or do I need to specify np_array? Also is xyzq clear or shall I specify that it's coordinates and charges?

File: stuff2sort/System.py (hunk 13 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Is self.list_geometry_initial better? Because it has also the data type

File: stuff2sort/System.py (hunk 14 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: I'm not sure, I think for people actually looking into the code it's enough to explain m1/m2 in the documentation. Also there is the comment above to describe the lists. I just can't think of a good variable name that describes m1/m2 atoms withour getting too long. 
For now I changed it to your suggested names until we have a final decision on that.

File: stuff2sort/System.py (hunk 15 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 16 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: To be honest, I don't know what corr means, but I will check with Jan to save time.

File: stuff2sort/System.py (hunk 17 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 18 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 19 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 20 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 21 above)
Comment (AlinaJansen) [reply stuff2sort/System.py:148]: Done

File: stuff2sort/System.py (hunk 22 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 23 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 24 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 25 above)
Comment (AlinaJansen) [reply stuff2sort/System.py:195]: # Find All Molecule Types And Their Amount
                else:
                    # Extract Lines With A String (Molecule Type, Group 1) And A Number (Molecule Amount, Group 2) Seperated By Whitespace
                    match = re.search(r"^(\S+)\s+(\d+)", line, flags=re.MULTILINE)
                    if match:
                        list_molecule.append([match.group(1), match.group(2)])

This is how I would comment this example. Is that alright?

File: stuff2sort/System.py (hunk 26 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 27 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 28 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 29 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done (:

File: stuff2sort/System.py (hunk 30 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 31 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: The _ in the end is for private variables, right? It just seems inconsistent as it's almost not used anywhere else in the code. Please explain why it's necessary here (:

File: stuff2sort/System.py (hunk 32 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 33 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 34 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 35 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 36 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 37 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 38 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 39 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 41 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 42 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 43 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 44 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: How do I write the return type if several variables are returned?

File: stuff2sort/System.py (hunk 16 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: I also saw that the other lists don't contain coordinates but only atom indices. I changed the names to self.list_atoms_*

File: stuff2sort/System.py (hunk 45 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 46 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Do you want to remove the whole function or just exclude the map into a seperate json file? I think we do need the function but putting the map in a json file would definitely improve readability (:

Comment (AlinaJansen) [reply stuff2sort/System.py]: I will keep the function and put the map into a json file

File: stuff2sort/System.py (hunk 12 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Keep array_xyzq

File: stuff2sort/System.py (hunk 14 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Add a comment in the code that more explanation is in the documenation

File: stuff2sort/System.py (hunk 31 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Just int_count

File: stuff2sort/System.py (hunk 14 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 31 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: done

File: stuff2sort/System.py (hunk 16 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Changed linkcorrlist to list_link_correction, it contains atom pairs for which Coulomb potential needs to be excluded.

File: stuff2sort/System.py (hunk 46 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Done

File: stuff2sort/System.py (hunk 44 above)
Comment (AlinaJansen) [reply stuff2sort/System.py]: Used tuple as return type, evaluata again later

**Review by FlorianAnders — APPROVED** [review]
Reviewed System.py
